### ISSUE TITLE
org-goto navigation keys do not work

### ISSUE THREAD

**Issue by bitozoid** [syl20bnr/spacemacs#15458]
#### Description :octocat:
Using vi keybinding, org-goto keyboard navigation does not work.

#### Reproduction guide :beetle:
- Start Emacs (spacemacs with vi keybindings).
- Open an org buffer.
- Run `C-c C-j` (org-goto).
- Navigate the org buffer with transient shortcuts.
- Try to jump to location with `RET`.

If `org-goto-auto-isearch` is `t`:
- Try to do a direct isearch by typing some word.

If `org-goto-auto-isearch` is `nil`:
- Try to navigate using [org-goto navigation keys](https://orgmode.org/manual/Motion.html).

*Observed behaviour:* :eyes: :broken_heart:

- `RET` does not jump to location
- direct isearch / keyboard navigation are not working either

*Expected behaviour:* :heart: :smile:

- `RET` to be jump to location
- direct isearch to work
- navigation using org-goto keys 

#### System Info :computer:
- OS: gnu/linux
- Emacs: 27.2
- Spacemacs: 0.999.0
- Spacemacs branch: develop (rev. 5c0650282)
- Graphic display: nil
- Distribution: spacemacs
- Editing style: vim
- Completion: helm
- Layers:
```elisp
(emacs-lisp git helm markdown multiple-cursors
            (org :variables org-agenda-span 'day org-agenda-skip-deadline-if-done t org-enable-reveal-js-support t org-enable-roam-support t org-enable-org-journal-support t org-roam-v2-ack t org-roam-directory
                 (file-truename "~/orgmatter/zettel")
                 org-journal-dir "~/orgmatter/journal" org-journal-date-format "%Y%m%d - %A, %e %B %Y" org-journal-file-header "#+STARTUP: indent show2levels" org-journal-file-format "%Y/%m/%Y%m%d.j.org" org-journal-carryover-items "TODO=\"WIP\"|TODO=\"NEXT\"|TODO=\"READY\"|TODO=\"SOMEDAY\"|TODO=\"WAITING\"" org-journal-enable-agenda-integration t org-want-todo-bindings t org-id-link-to-org-use-id 'create-if-interactive-and-no-custom-id)
            treemacs python yaml emoji xclipboard auto-completion pdf bibtex
            (mu4e :variables mu4e-use-maildirs-extension t mu4e-org-support t))
```
- System configuration features: XPM JPEG TIFF GIF PNG RSVG CAIRO SOUND DBUS GCONF GLIB NOTIFY INOTIFY GNUTLS LIBXML2 FREETYPE HARFBUZZ LIBOTF ZLIB TOOLKIT_SCROLL_BARS GTK3 X11 XDBE XIM MODULES THREADS LIBSYSTEMD JSON PDUMPER LCMS2 GMP

#### Backtrace :paw_prints:

No error no backtrace.
